apps/dashboard/src/context/environment/environment-provider.tsx

```diff
@@ -35,18 +32,51 @@ function selectEnvironment(environments?: IEnvironment[], selectedEnvironmentId?
 
 export function EnvironmentProvider({ children }: { children: React.ReactNode }) {
   const { currentOrganization } = useAuth();
+  const navigate = useNavigate();
+  const { pathname } = useLocation();
+  const { environmentId: paramsEnvironmentId } = useParams<{ environmentId?: string }>();
   const [currentEnvironment, setCurrentEnvironment] = useState<IEnvironment | null>(null);
 
+  const switchEnvironmentInternal = useCallback(
+    (allEnvironments: IEnvironment[], environmentId?: string | null) => {
+      const selectedEnvironment = selectEnvironment(allEnvironments, environmentId);
+      setCurrentEnvironment(selectedEnvironment);
+      const newEnvironmentId = selectedEnvironment._id;
+      const isNewEnvironmentDifferent = paramsEnvironmentId !== selectedEnvironment._id;
```

Comment:
Wouldn't the environment be always different on switching?

Reply:
@SokratisVidros no, this is for case when someone will type the wrong `environmentId` or just typed `asdf`